### DISCUSSION TITLE
Programming EPM1270

### DISCUSSION THREAD

**Discussion by anarsoul** [UzixLS/zx-sizif-512#48] [Q&A] [answered]
Do I understand correctly that JTAG interface should be working just with EPM1270 and the header on the board? Does it require any other components?

I bought some EPM1270 from aliexpress and I'd like to test that they can be programmed without building whole board.

**Comment by UzixLS** [comment]
You also need JTAG pullpup resistors and power circuit (including 1117-3.3 LDO)

**Reply by anarsoul** [reply]
Hmm, I don't see JTAG pull-ups on the schematics for E1

**Reply by UzixLS** [reply]
Whoops, you're right, they are internal in EPM1270

**Reply by anarsoul** [reply]
Does it mean that if I use bench power supply for 3.3V, I need only CPLD and JTAG header on the board?

**Reply by UzixLS** [reply, marked as answer]
Yes. But I would recommend to put capacitors on 3.3V for stable operation.

**Reply by anarsoul** [reply]
Just FYI, I was able to program CPLD with just CPLD and the header on the board, 3.3V were supplied via bench power supply.
